graylog2-web-interface/src/components/grok-patterns/GrokPatterns.jsx

```diff
@@ -87,26 +88,30 @@ const GrokPatterns = React.createClass({
             your own manually or import a whole list of patterns from a so called pattern file.
           </span>
           {null}
-          <span>
-            <BulkLoadPatternModal onSuccess={this.loadData} />
-            <EditPatternModal id={''} name={''} pattern={''} create
-                              reload={this.loadData}
-                              savePattern={this.savePattern}
-                              validPatternName={this.validPatternName} />
-          </span>
+          <IfPermitted permissions="inputs:edit">
+            <span>
+              <BulkLoadPatternModal onSuccess={this.loadData} />
+              <EditPatternModal id={''} name={''} pattern={''} create
+                                reload={this.loadData}
+                                savePattern={this.savePattern}
+                                validPatternName={this.validPatternName} />
+            </span>
+          </IfPermitted>
         </PageHeader>
 
         <Row className="content">
           <Col md={12}>
-            <DataTable id="grok-pattern-list"
-                       className="table-striped table-hover"
-                       headers={headers}
-                       headerCellFormatter={this._headerCellFormatter}
-                       sortByKey={'name'}
-                       rows={this.state.patterns}
-                       dataRowFormatter={this._patternFormatter}
-                       filterLabel="Filter patterns"
-                       filterKeys={filterKeys} />
+            <IfPermitted permissions="inputs:read">
```

Comment:
This `DataTable` component also renders `delete` and `edit` buttons for each grok pattern. They don't seem to work, but it feels "wrong". I guess we should either disable (or remove) those buttons, or not render the table unless the user can edit it.